File: leetcode_problems/cpp/lc2269_find_k_beauty_of_a_number.cpp
#include <string>
#include <cctype>
using std::string;
using std::to_string;

class Solution
{
public:
    int divisorSubstrings(int num, int k)
    {
        string numStr = to_string(num);
        int n = numStr.size();
        int count = 0;

        for (int i = 0; i <= n - k; i++)
        {
            string subStr = numStr.substr(i, k);
            int subNum = std::stoi(subStr);
            if (subNum != 0 && num % subNum == 0)
            {
                count++;
            }
        }

        return count;
    }
};

File: leetcode_problems/typescript/src/lc2269_find_k_beauty_of_a_number.ts
function divisorSubstrings(num: number, k: number): number {
  let numStr = num.toString();
  let n = numStr.length;
  let count = 0;
  for (let i = 0; i <= n - k; i++) {
    let subStr = numStr.slice(i, i + k);
    let subNum = Number(subStr);
    if (subNum !== 0 && num % subNum === 0) {
      count++;
    }
  }
  return count;
}
